crates/circuit/src/circuit_data.rs

```diff
@@ -504,6 +504,25 @@ impl CircuitData {
         Ok(())
     }
 
+    /// Lists all operations along with the qubits involved in those operations.
+    ///
+    /// Returns:
+    ///     list.
+    #[pyo3(signature = ())]
+    pub fn operations_with_qubits(&self, py: Python<'_>) -> PyResult<Py<PyAny>> {
+        let mut ret: Vec<(Option<String>, Option<String>)> = vec![];
```

Comment:
I think I think the return type here is a bit off. What I think would be useful here is to return either a vec or hashmap of qubit indices that map to the instruction counts on them. For example something like:
```
[
    {"h": 20, "cx" 2, "s": 10},
    {"cx": 2},
]
```
for a circuit that has 2 qubits with 20 hadamards, 10 s gates, and 2 cx gates on qubit 0 and qubit 1 has 2 cx gates.

Reply:
hi @mtreinish , I'm a bit stuck here, how could I get this information (the qubit numbers) without having to get substrings on the string gotten from the Qubit objects from 'Some(self.qubits.get(*b).unwrap().to_string()'? Or should I go ahead and get the substrings to obtain the qubits and populate the HasMaps from that?

Reply:
Oh, that's not too difficult for any given `PackedInstruction` object `instr` you get while iterating over the circuit data you can call `CircuitData::get_qargs(instr.qubits)` which will return a `&[Qubit]` object. The `Qubit` object is a [newtype](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) `u32`. So you can do `Qubit.0 as usize` to get the index into a vec for any `Qubit`.

Reply:
Ok, will follow that procedure. Thanks